### PR TITLE
Release-process.md: rebase and merge

### PR DESCRIPTION
**Pull request by jakkab**
**Description**

Changes proposed in this pull request:

- Use the "rebase and merge" option to merge PRs to the release branch.

#294

### CODE REVIEW

File: guidelines/releases/release-process.md [hunk 1]
```diff
@@ -201,7 +201,9 @@ Follow these steps to create a release:
 6. If you detect any problems with the release, such as failing tests, wait for the fix that can be delivered either on a PR or cherry-picked to the PR from the `master` branch.  
 	Prow triggers the jobs again. Rerun manual jobs as described in **step 5**.
 
-7. After all checks pass, merge the PR.
+7. After all checks pass, merge the PR, using the `rebase and merge` option.
+    > **CAUTION:** By default, the `rebase and merge` option is disabled. Ask one of the `kyma-project/kyma` repository admins to enable it.
```
Comment (kazydek) [inline guidelines/releases/release-process.md]:
```suggestion
    > **CAUTION:** By default, the `rebase and merge` option is disabled. Contact one of the `kyma-project/kyma` repository admins to enable it.
```

File: guidelines/releases/release-process.md [hunk 2]
```diff
@@ -201,7 +201,9 @@ Follow these steps to create a release:
...
+    
 	> **NOTE:** To merge the PR to the release branch, you must receive approvals from all teams.
```
Comment (kazydek) [inline guidelines/releases/release-process.md]:
```suggestion
```

File: guidelines/releases/release-process.md [hunk 3]
```diff
@@ -201,7 +201,9 @@ Follow these steps to create a release:
 6. If you detect any problems with the release, such as failing tests, wait for the fix that can be delivered either on a PR or cherry-picked to the PR from the `master` branch.  
 	Prow triggers the jobs again. Rerun manual jobs as described in **step 5**.
 
-7. After all checks pass, merge the PR.
+7. After all checks pass, merge the PR, using the `rebase and merge` option.
```
Comment (kazydek) [inline guidelines/releases/release-process.md]:
```suggestion
7. After all checks pass, merge the PR, using the `rebase and merge` option. To merge the PR to the release branch, you must receive approvals from all teams.
```

File: guidelines/releases/release-process.md [hunk 4]
```diff
@@ -201,8 +201,9 @@ Follow these steps to create a release:
 6. If you detect any problems with the release, such as failing tests, wait for the fix that can be delivered either on a PR or cherry-picked to the PR from the `master` branch.  
 	Prow triggers the jobs again. Rerun manual jobs as described in **step 5**.
 
-7. After all checks pass, merge the PR.
-	> **NOTE:** To merge the PR to the release branch, you must receive approvals from all teams.
+7. After all checks pass, merge the PR, using the `rebase and merge` option. To merge the PR to the release branch, you must receive approvals from all teams.
+    > **CAUTION:** By default, the `rebase and merge` option is disabled. Contact one of the `kyma-project/kyma` repository admins to enable it.
```
Comment (pkosiec) [inline guidelines/releases/release-process.md]: to enable it. Remember about disabling it after merge.

**Pull request merged**